File: library/src/main/java/leavesc/hello/basis/widget/LoadingDialog.java
package leavesc.hello.basis.widget;

import android.app.Dialog;
import android.content.Context;
import android.support.annotation.StringRes;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;
import android.widget.TextView;

import leavesc.hello.basis.R;

/**
 * 作者：leavesC
 * 时间：2019/3/23 11:43
 * 描述：
 * GitHub：https://github.com/leavesC
 * Blog：https://www.jianshu.com/u/9df45b87cfdf
 */
public class LoadingDialog extends Dialog {

    private ImageView iv_loading;

    private TextView tv_hint;

    private Animation animation;

    public LoadingDialog(Context context) {
        super(context, R.style.LoadingDialogTheme);
        setContentView(R.layout.dialog_loading);
        iv_loading = findViewById(R.id.iv_loading);
        tv_hint = findViewById(R.id.tv_hint);
        animation = AnimationUtils.loadAnimation(context, R.anim.loading_dialog);
    }

    public void show(String hintText, boolean cancelable, boolean canceledOnTouchOutside) {
        setCancelable(cancelable);
        setCanceledOnTouchOutside(canceledOnTouchOutside);
        tv_hint.setText(hintText);
        iv_loading.clearAnimation();
        iv_loading.startAnimation(animation);
        show();
    }

    public void show(@StringRes int hintTextRes, boolean cancelable, boolean canceledOnTouchOutside) {
        show(getContext().getString(hintTextRes), cancelable, canceledOnTouchOutside);
    }

    @Override
    public void cancel() {
        super.cancel();
        animation.cancel();
        iv_loading.clearAnimation();
    }

    @Override
    public void dismiss() {
        super.dismiss();
        animation.cancel();
        iv_loading.clearAnimation();
    }

}

File: library/src/main/java/leavesc/hello/basis/widget/dialog/LoadingDialog.kt
package leavesc.hello.basis.widget.dialog

import android.app.Dialog
import android.content.Context
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.ImageView
import android.widget.TextView
import androidx.annotation.StringRes
import leavesc.hello.basis.R

/**
 * 作者：leavesC
 * 时间：2019/6/21 10:51
 * 描述：
 */
class LoadingDialog(context: Context) : Dialog(context, R.style.LoadingDialogTheme) {

    private val iv_loading: ImageView

    private val tv_hint: TextView

    private val animation: Animation

    init {
        setContentView(R.layout.dialog_loading)
        iv_loading = findViewById(R.id.iv_loading)
        tv_hint = findViewById(R.id.tv_hint)
        animation = AnimationUtils.loadAnimation(context, R.anim.loading_dialog)
    }

    fun show(hintText: String, cancelable: Boolean, canceledOnTouchOutside: Boolean) {
        setCancelable(cancelable)
        setCanceledOnTouchOutside(canceledOnTouchOutside)
        tv_hint.text = hintText
        iv_loading.clearAnimation()
        iv_loading.startAnimation(animation)
        show()
    }

    fun show(@StringRes hintTextRes: Int, cancelable: Boolean, canceledOnTouchOutside: Boolean) {
        show(context.getString(hintTextRes), cancelable, canceledOnTouchOutside)
    }

    override fun cancel() {
        super.cancel()
        animation.cancel()
        iv_loading.clearAnimation()
    }

    override fun dismiss() {
        super.dismiss()
        animation.cancel()
        iv_loading.clearAnimation()
    }

}